src/main/base/Ball.java

```diff
@@ -69,8 +69,28 @@ public void move() { // ball moves
     }
 
     // ball hit something
-    public void hit(float d, int player) {
-        currentSpeedY = hitSpeedY - 1; // handle y speed
+    public void hit(float d, int player,boolean ifSmash,int y) {
+        if(player==1){
+            if(ifSmash==true&&y<500 && d>0.0f){
+                currentSpeedY = -hitSpeedY - 5; // smash the ball
+                //System.out.println(d);
+            }else{
+                currentSpeedY = hitSpeedY - 1; // handle y speed
+                //System.out.println(d);
+            }
+        }else if(player==2){
+            if(ifSmash==true&&y<500 && d<0.0f){
+                currentSpeedY = -hitSpeedY - 5; // smash the ball
+                //System.out.println(d);
+            }else{
+                currentSpeedY = hitSpeedY - 1; // handle y speed
+                System.out.printf("ifSmash%b\n",ifSmash);
+                System.out.printf("y%d\n",y);
+                System.out.printf("d%f\n",d);
+            }
+        }
+        
+        
```

Comment:
Some issues that are not too important, but I think modifying these details can make the playing experience better

* You have to judge the various conditions of smash according to the direction of the key (ex. for player1, the X value when VK_A is pressed will be smaller than VK_D), so there have 4 direction you have to judge
* You don't need to calculate two player, they are basically the same, you just need calculate one of them, and reverse the other one
* You don't need to care if ball higher or lower then border, this will judge by `collision` in `main.panel.Game`